src/celeritas/global/detail/AlongStepActionImpl.hh

```diff
@@ -145,8 +145,12 @@ inline CELER_FUNCTION void along_step_track(CoreTrackView const& track)
         const auto& urban_data = phys.urban_data();
 
         // Replace step with actual geometry distance traveled
-        auto msc_step_result      = track.make_physics_step_view().msc_step();
-        msc_step_result.geom_path = geo_step;
+        auto msc_step_result = track.make_physics_step_view().msc_step();
+        if (msc_step_result.geom_path != geo_step)
+        {
+            msc_step_result.geom_path   = geo_step;
+            msc_step_result.geo_limited = true;
+        }
```

Comment:
Instead of using a floating point comparison, use the "action ID" which more precisely indicates whether the boundary was hit.

```suggestion
        auto msc_step_result = track.make_physics_step_view().msc_step();
        msc_step_result.geo_limited
            = (step_limit.action == track.boundary_action());
        if (msc_step_result.geo_limited)
        {
            msc_step_result.geom_path = geo_step;
        }
```

Reply:
Actually, instead of modifying the `msc_step_result` struct (that bool doesn't need to be stored in memory) just make a local bool and add a parameter to the `MscScatter` class.

Reply:
Okay, add a local bool and pass it to the UrbanMscScatter as an argument - which was my original choice, but did not want to change the interface.